### PR TITLE
feat: introduce arbitrum network

### LINKED ISSUE #3571: [Demand] Integrate Arbitrum
## Catalogue

https://offchainlabs.com/

as one of the EVM compatible scaling solution 

- [ ] UI/UX
- [ ] Mask Core Feature
- [ ] Mask Plugin Feature
- [ ] Security/Cryptography Feature
- [ ] Other
  - /_List it here_/

### PR DESCRIPTION
**Pull request by guanbinrui**
Learn more from: https://developer.offchainlabs.com/docs/developer_quickstart.

## Contracts

+ [x] https://github.com/makerdao/multicall
+ [x] https://github.com/wbobeirne/eth-balance-checker

closes #3571

### CODE REVIEW

**Review by UncleBill — APPROVED** [review]

File: .vscode/schema/constants.json [hunk 1]
```diff
@@ -19,7 +19,7 @@
     "minProperties": 1,
     "definitions": {
         "chain-id": {
-            "enum": ["Mainnet", "Ropsten", "Rinkeby", "Gorli", "Kovan", "BSC", "BSCT", "Matic", "Mumbai"]
+            "enum": ["Mainnet", "Ropsten", "Rinkeby", "Gorli", "Kovan", "BSC", "BSCT", "Matic", "Mumbai", "ARBITRUM"]
```
Comment (guanbinrui) [inline .vscode/schema/constants.json:22]:
```suggestion
            "enum": ["Mainnet", "Ropsten", "Rinkeby", "Gorli", "Kovan", "BSC", "BSCT", "Matic", "Mumbai", "Arbitrum"]
```

File: packages/constants/compile-constants.ts [hunk 2]
```diff
@@ -4,7 +4,7 @@ import { run } from '../cli/utils'
 
 const DATA_PATH = path.join(__dirname, 'data')
 
-const chainNames = ['Mainnet', 'Ropsten', 'Rinkeby', 'Kovan', 'Gorli', 'BSC', 'BSCT', 'Matic', 'Mumbai']
+const chainNames = ['Mainnet', 'Ropsten', 'Rinkeby', 'Kovan', 'Gorli', 'BSC', 'BSCT', 'Matic', 'Mumbai', 'ARBITRUM']
```
Comment (guanbinrui) [inline packages/constants/compile-constants.ts:7]:
```suggestion
```suggestion
            "enum": ["Mainnet", "Ropsten", "Rinkeby", "Gorli", "Kovan", "BSC", "BSCT", "Matic", "Mumbai", "Arbitrum", "Arbitrum_Rinkeby"]
```
```

File: packages/web3-constants/data/gitcoin.json [hunk 3]
```diff
@@ -41,6 +47,8 @@
         "BSC": 5,
         "BSCT": 5,
         "Matic": 5,
-        "Mumbai": 5
+        "Mumbai": 5,
+        "Arbitrum": 0,
+        "Arbitrum_Rinkeby": 0
```
Comment (guanbinrui) [inline packages/web3-constants/data/gitcoin.json:52]:
```suggestion
        "Arbitrum": 5,
        "Arbitrum_Rinkeby": 5
```

**Review by septs — DISMISSED** [review]

**Review by Jack-Works — DISMISSED** [review]

File: packages/web3-constants/compile-constants.ts [hunk 4]
```diff
@@ -2,13 +2,26 @@ import { promises as fs } from 'fs'
 import path from 'path'
 import { run } from '../cli/utils'
 
+// const ROOT_PATH = path.join(__dirname, '..', '..')
```
Comment (septs) [inline packages/web3-constants/compile-constants.ts:5]: remove the line

**Review by Jack-Works — APPROVED** [review]

File: packages/web3-constants/compile-constants.ts (hunk 4 above)
Comment (septs) [reply packages/web3-constants/compile-constants.ts:5]:
```suggestion
```

**Review by septs — APPROVED** [review]

File: packages/web3-constants/data/ito.json [hunk 5]
```diff
@@ -74,7 +74,7 @@
         "BSCT": "",
         "Matic": "https://api.thegraph.com/subgraphs/name/dimensiondev/mask-ito-polygon",
         "Mumbai": "",
-        "Arbitrum": "",
+        "Arbitrum": "https://thegraph.com/legacy-explorer/subgraph/dimensiondev/mask-ito-arbitrum-one",
```
Comment (septs) [inline packages/web3-constants/data/ito.json:77]: why not is `https://api.thegraph.com`?

**Review by Jack-Works — APPROVED** [review]

**Review by septs — CHANGES_REQUESTED** [review]
plz use https://imageoptim.com pre-reduce image resources

**Review by septs — APPROVED** [review]

**Review by septs — APPROVED** [review]

**Comment by Jack-Works** [comment]
ci/circleci: build — Your tests failed on CircleCI

**Review by septs — APPROVED** [review]

**Review by septs — APPROVED** [review]

File: packages/maskbook/src/components/shared/NetworkIcon.tsx [hunk 6]
```diff
@@ -15,22 +15,20 @@ export interface NetworkIconProps extends withClasses<'icon'> {
     networkType?: NetworkType
 }
 
-const EthereumIcon = new URL('../../resources/wallet-network-icon/ethereum.png', import.meta.url).toString()
-const BinanceIcon = new URL('../../resources/wallet-network-icon/binance.png', import.meta.url).toString()
-const PolygonIcon = new URL('../../resources/wallet-network-icon/polygon.png', import.meta.url).toString()
+const icons: EnumRecord<NetworkType, string> = {
```
Comment (Jack-Works) [inline packages/maskbook/src/components/shared/NetworkIcon.tsx:18]: Please use `Record` and remove `EnumRecord`. Thanks. (These two types are identical.)

**Review by Jack-Works — APPROVED** [review]

File: packages/maskbook/src/components/shared/NetworkIcon.tsx (hunk 6 above)
Comment (guanbinrui) [reply packages/maskbook/src/components/shared/NetworkIcon.tsx]: REMOVED.

**Review by septs — DISMISSED** [review]

File: packages/maskbook/src/components/shared/NetworkIcon.tsx [hunk 7]
```diff
@@ -15,22 +15,20 @@ export interface NetworkIconProps extends withClasses<'icon'> {
...
-        case NetworkType.Polygon:
-            return <Image height={size} width={size} src={PolygonIcon} className={classes.icon} />
-        default:
-            return null
-    }
+    if (!networkType) return null
+
+    return icons[networkType] ? (
+        <Image height={size} width={size} src={icons[networkType]} className={classes.icon} />
+    ) : null
```
Comment (septs) [inline packages/maskbook/src/components/shared/NetworkIcon.tsx:33]:
```suggestion
    if (!(networkType in icons)) return null

    return <Image height={size} width={size} src={icons[networkType]} className={classes.icon} />
```

**Pull request merged**